src/hotspot/share/opto/escape.hpp

```diff
@@ -668,4 +668,13 @@ inline FieldNode::FieldNode(ConnectionGraph *CG, Node* n, EscapeState es, int of
   _has_unknown_base(false) {
 }
 
+inline bool typeCannotEscape(const TypeKlassPtr *ptr, const Compile *c) {
```

Comment:
I would leave EscapeAnalysis alone. I understand that you want to factor out the common logic. in my understanding, PEA has its own logic. Touching the established EA will add extra works for future reviewers.